src/java.base/share/classes/java/lang/foreign/Linker.java

```diff
@@ -260,7 +260,9 @@
  * <li>each member layout in {@code G.memberLayouts()} is either a padding layout or
  *     a layout supported by {@code NL}, and</li>
  * <li>{@code G} does not contain padding other than what is strictly required to align
- *      its non-padding layout elements, or to satisfy (2).</li>
+ *      its non-padding layout elements, or to satisfy (2), and</li>
+ * <li>{@code G} is {@code G.memberLayouts().isEmpty()} or (at the same time) not all
```

Comment:
I think @JornVernee mentioned that empty layouts are ok and/or accepted by C compilers. We just don't know what to do with padding-only groups.